app/src/main/java/android/coding/interview/makeitawesome/fragment/PicturesFragment.java

```diff
@@ -19,19 +30,78 @@
 public class PicturesFragment extends Fragment {
 
     public static Fragment newInstance() {
-        return new PicturesFragment();
+        PicturesFragment pictures = new PicturesFragment();
+        pictures.setRetainInstance(true);
+        return pictures;
     }
 
+    RecyclerView rv;
+    ProgressDialog mProgressDialog;
+    List<Album> mData;
+    boolean loading = false;
+    private TinyBus mBus;
+
     @Nullable
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        RecyclerView rv = (RecyclerView) inflater.inflate(R.layout.pictures_list_fragment, container, false);
-        setupRecyclerView(rv);
+        rv = (RecyclerView) inflater.inflate(R.layout.pictures_list_fragment, container, false);
+
+        if (mProgressDialog != null && mProgressDialog.isShowing())
+            mProgressDialog.dismiss();
+
+        mProgressDialog = new ProgressDialog(container.getContext());
+        mProgressDialog.setIndeterminate(true);
+        mProgressDialog.setMessage("Loading...");
+        mProgressDialog.show();
+
+        mBus = TinyBus.from(getActivity());
+
+        if (mData == null) {
+            if (!loading) {
+                RestClient restClient = RestClient.getInstance();
+                restClient.getPhotos(mBus);
+                loading = true;
+            }
+        } else {
+            drawList();
```

Comment:
Dialog is hidden in drawlist(), but if mData was != null, he shouldn't have ever shown it. I get more worried about this kind of thins. A medium-good programmer that moreover is trying to impress you... I want to think I wouldn't write such a spaghetti code for something relatively simple.